modules/ServerAPI/src/server/routes/FileServices.coffee

```diff
@@ -200,15 +459,134 @@ setupRoutes = (app, loginRoutes, requireLogin) ->
 
 exports.deleteFile = (req, resp) ->
 	# Not implemented currently. We added this route in order to fully support the jquery file upload plugin
 	# which requires a delete route in order to function correctly with maxNumberOfFiles.
 	# See details here: https://github.com/mcneilco/acas/pull/1085
 	console.log 'Got DELETE request for data file: ' + req.params[0] + ' from user: ' + req.user.username
 	console.log 'Not deleting file, just returning 200 OK'
-	resp.send 200
\ No newline at end of file
+	resp.send 200
+
+
+exports.moveDataFiles = (req, resp) ->
+	deleteSourceFileOnSuccess = if req.query?.deleteSourceFileOnSuccess? then req.query.deleteSourceFileOnSuccess == "true" else true
+	files = await exports.moveDataFilesInternal(req.body, deleteSourceFileOnSuccess)
+	resp.json files
+
+exports.moveDataFilesInternal = (files, deleteSourceFileOnSuccess) ->
+	return await exports.fileHandler.moveFiles(files, deleteSourceFileOnSuccess)
+
+
+exports.migrateCmpdRegBulkLoaderFilesToSubfolders = () ->
+	console.log "About to migrate cmpdReg bulk loader files to subfolders"
+	cmpdRegBulkLoaderRoutes = require('./CmpdRegBulkLoaderRoutes.js')
+	bulkLoadFiles = await cmpdRegBulkLoaderRoutes.getBulkloadFilesInternal()
+	console.log "There are #{bulkLoadFiles.length} bulk load files in the DB"
+	bulkLoadSubFolderFiles = await exports.localFileHandler.listFiles(cmpdRegBulkLoaderRoutes.BULKLOAD_SUB_FOLDER, false)
+	console.log "There are #{bulkLoadSubFolderFiles.length} bulk load files in the local bulkload subfolder"
+	if exports.fileHandler instanceof ExternalFileHandler
+		externalBulkLoadSubFolderFiles = bulkLoadSubFolderFiles.concat(await exports.fileHandler.listFiles(cmpdRegBulkLoaderRoutes.BULKLOAD_SUB_FOLDER, false))
+		console.log "There are #{externalBulkLoadSubFolderFiles.length} bulk load files in the external bulkload subfolder"
+		# Organize by bulk load file id which is the first subfolder under the registered folder
+		storedFilesByID = {}
+		for file in externalBulkLoadSubFolderFiles
+			fileID = path.basename(path.dirname(file.path))
+			if !storedFilesByID[fileID]
+				storedFilesByID[fileID] = []
+			storedFilesByID[fileID].push(file)
+
+	# For each of the bulkLoadFiles in the DB check if the file is in the storedFilesByID dict
+	filesToMove = []
+	for bulkLoadFile in bulkLoadFiles
+		registeredFilesFolder = path.join(cmpdRegBulkLoaderRoutes.REGISTERED_FOLDER, bulkLoadFile.id.toString())
+		registeredStoredFiles = await exports.localFileHandler.listFiles(registeredFilesFolder)
+		fileNameNoExtension = path.basename(bulkLoadFile.fileName, path.extname(bulkLoadFile.fileName))
+		
+		# Find all bulkLoadSubFolderFiles files that start with the fileNameNoExtension
+		matchingBulkLoadSubFolderFiles = bulkLoadSubFolderFiles.filter (file) ->
+			path.basename(file.path).startsWith(fileNameNoExtension)
+
+		if matchingBulkLoadSubFolderFiles.length == 0 && registeredStoredFiles.length == 0
```

Comment:
With these variable names it's very hard for me to tell what this case means. No local files and no registered files?